clang/test/SemaSYCL/check-work-group-size-hint-device.cpp

```diff
@@ -17,7 +17,8 @@
 
 // Produce a conflicting attribute warning when the args are different.
 [[sycl::work_group_size_hint(4, 1, 1)]] void f3();    // expected-note {{previous attribute is here}}
-[[sycl::work_group_size_hint(1, 1, 32)]] void f3() {} // expected-warning {{attribute 'work_group_size_hint' is already applied with different arguments}}
+[[sycl::work_group_size_hint(1, 1, 32)]] void f3() {} // expected-warning {{attribute 'work_group_size_hint' is already applied with different arguments}} \
```

Comment:
The requirements for this warning are a bit unclear to me. Do you want a warning generated for non-device functions? I also don't fully understand what the behavior should be for function declarations (and not definitions). It feels odd to me that the warning is fired for L20 and not L19.

Reply:
I want to issue a warning for any function that uses these attributes when it is not a kernel.

Reply:
So you suggest that warning should be fired at the function declaration?

Reply:
>>I want to issue a warning for any function that uses these attributes when it is not a kernel.

What happens now when the attribute is applied to a non-device function i.e. a host function? What is the required behavior in this case?

>>So you suggest that warning should be fired at the function declaration?

I'm not sure to be honest. If we're emitting it only for device functions, maybe current behavior is ok but I am not 100% sure.

Reply:
>>I want to issue a warning for any function that uses these attributes when it is not a kernel.
>>>>What happens now when the attribute is applied to a non-device function i.e. a host function? What is the required behavior in this case?

No need to answer this question. I see you emit it for all functions.